<?xml version="1.0" ?>
<robot name="eSAUl" xmlns:xacro="http://www.ros.org/wiki/xacro">
  
  <xacro:include filename="$(find eSAUl_ad_helll)/urdf/wheel.xacro" />
  <link name="link_chassis">

    <inertial>
      	<mass value="5"/>
      	<origin rpy="0 0 0" xyz="0 0 0"/>
      	<inertia ixx="0.0395416666667" ixy="0" ixz="0" iyy="0.106208333333" iyz="0" izz="0.106208333333"/>
    </inertial>

    <!-- body -->
    <collision name="collision_chassis">
      <geometry>
        <box size="0.45 0.22 0.13"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.45 0.22 0.13"/>
      </geometry>
      <material name="blue"/>
    </visual>

    
  </link>

	<xacro:link_wheel name="link_right_wheel_1" />
	<xacro:joint_wheel name="joint_right_wheel_1" child="link_right_wheel_1" origin_xyz="0.13 0.13 -0.05" />
  
	<xacro:link_wheel name="link_left_wheel_1" />
	<xacro:joint_wheel name="joint_left_wheel_1" child="link_left_wheel_1" origin_xyz="0.13 -0.13 -0.05" />

	<xacro:link_wheel name="link_right_wheel_2" />
	<xacro:joint_wheel name="joint_right_wheel_2" child="link_right_wheel_2" origin_xyz="-0.17 0.13 -0.05" />
  
	<xacro:link_wheel name="link_left_wheel_2" />
	<xacro:joint_wheel name="joint_left_wheel_2" child="link_left_wheel_2" origin_xyz="-0.17 -0.13 -0.05" />
	
	<xacro:link_wheel name="link_right_wheel_3" />
	<xacro:joint_wheel name="joint_right_wheel_3" child="link_right_wheel_3" origin_xyz="0.03 0.13 -0.05" />
  
	<xacro:link_wheel name="link_left_wheel_3" />
	<xacro:joint_wheel name="joint_left_wheel_3" child="link_left_wheel_3" origin_xyz="0.03 -0.13 -0.05" />

	<xacro:link_wheel name="link_right_wheel_4" />
	<xacro:joint_wheel name="joint_right_wheel_4" child="link_right_wheel_4" origin_xyz="-0.07 0.13 -0.05" />
  
	<xacro:link_wheel name="link_left_wheel_4" />
	<xacro:joint_wheel name="joint_left_wheel_4" child="link_left_wheel_4" origin_xyz="-0.07 -0.13 -0.05" />

  <!--
<gazebo>
  <plugin name="differential_drive_controller" filename="libgazebo_ros_diff_drive.so">
    <legacyMode>false</legacyMode>
    <alwaysOn>true</alwaysOn>
    <publishWheelTF>true</publishWheelTF>
    <publishTf>1</publishTf>
    <publishWheelJointState>true</publishWheelJointState>
    <updateRate>100.0</updateRate>
    <leftJoint>joint_right_wheel_2</leftJoint>
    <rightJoint>joint_left_wheel_1</rightJoint>
    <wheelSeparation>0.22</wheelSeparation>
    <wheelDiameter>0.08</wheelDiameter>
    <wheelAcceleration>1.0</wheelAcceleration>
    <torque>20</torque>
    <commandTopic>/part2_cmr/cmd_vel</commandTopic>
    <odometryTopic>odom</odometryTopic>
    <odometryFrame>odom</odometryFrame>
    <robotBaseFrame>link_chassis</robotBaseFrame>
  </plugin>
</gazebo>
 -->
<gazebo>
    <!-- <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
      <robotNamespace>/</robotNamespace>
    </plugin> -->

    <!-- Simulated vehicle interface -->
    <plugin name="ModelPlugin" filename="librobot_gazebo.so" >
      <robotName>eSAUl</robotName>
      <wheelbase>link_chassis</wheelbase>
      <joint_right_1>joint_right_wheel_1</joint_right_1>
      <joint_left_1>joint_left_wheel_1</joint_left_1>
      <joint_right_2>joint_right_wheel_2</joint_right_2>
      <joint_left_2>joint_left_wheel_2</joint_left_2>
      <joint_right_3>joint_right_wheel_3</joint_right_3>
      <joint_left_3>joint_left_wheel_3</joint_left_3>	
      <joint_right_4>joint_right_wheel_4</joint_right_4>
      <joint_left_4>joint_left_wheel_4</joint_left_4>	
    </plugin>
</gazebo>

</robot>

